### DISCUSSION TITLE
After upgrade to v3: Changing x-data in DOM makes Alpine component completely unresponsive

### DISCUSSION THREAD

**Discussion by plorenz-etes** [alpinejs/alpine#3918] [1. Help]
So I have finally managed to upgrade my webapp to Alpine v3, however the upgrade causes a strange issue. To even make it worse, I was unable to reproduce the issue in an example project, but maybe someone has a suggestion:

I have an Alpine component that stops working as soon as the `x-data` attribute inside the DOM is modified. At first, everything works perfectly fine, but after manually changing the `x-data` attribute using the browser's dev tools, the component stops working. Changing any of its variables does nothing and also events won't be fired anymore.

Of course, changing the `x-data` attribute using the dev tools isn't a real use-case - this is just a simplification of the real problem: Livewire re-rendering a component and thus overwriting the attribute with other values.

It's even stranger that this doesn't happen with other Alpine components on the same page. It also didn't happen with Alpine v2.

Does anybody have a suggestion on how to resolve or debug this issue?

**Comment by ekwoka** [comment]
Are you using Livewire 3?

**Reply by plorenz-etes** [reply]
I tried both v2 and v3, but it happens with both :/

**Reply by ekwoka** [reply]
A codepen would be helpful

**Reply by plorenz-etes** [reply]
That's the problem - I would provide one if I could reproduce the issue, however it only occurs in one project and only in one of its components. If I should succeed to reproduce the issue, I'll let you know.

**Reply by ekwoka** [reply]
Then I'd expect it's something specific to do with something there.

Any error messages?

**Reply by plorenz-etes** [reply]
Unfortunately not :( Nothing is shown in the console and the same happens in any browser.
I also installed the AlpineJS devtools for Chrome, there I can toggle my Alpine variables and this works perfectly for every component, but after altering the `x-data` attribute, toggling the vars doesn't have any effect anymore.

**Reply by SimoTod** [reply]
Depending on what you are changing, you may be generating a brand new data scope whilst some of the other existing directives are watching for changes on the old one. In theory, you shouldn't change the x-data content manually too much but use entangle to share state between frontend and backend

**Reply by ekwoka** [reply]
That's what I would expect as well.

that everything is already hooked into the old components stack, specifically in hte context of nested components. Likely only directives immediately inside that data could be transferred.
